src/components/UserProfile/UserProfile.jsx

```diff
@@ -1,9 +1,135 @@
-import React from 'react'
+import useAuth from '../../contexts/AuthContext';
+import { db } from '../../config/firebase';
+import { doc, getDoc } from '@firebase/firestore';
+import { useEffect, useState } from 'react';
+import styles from './userprofile.module.css';
+import {
+	getStorage,
+	ref,
+	getDownloadURL,
+	uploadBytes,
+} from '@firebase/storage';
+import { toast } from 'react-hot-toast';
 
 const UserProfile = () => {
-  return (
-    <div>UserProfile</div>
-  )
-}
+	const { currentUser } = useAuth();
+	const [user, setUser] = useState();
 
-export default UserProfile
\ No newline at end of file
+	const [age, setAge] = useState();
+
+	const storage = getStorage();
+	const [userProfileImgURL, setUserProfileImgURL] = useState();
+
+	const getUserData = async () => {
+		const userRef = await doc(db, 'Users', currentUser.uid);
+		await getDoc(userRef).then((userSnapshot) => {
+			setUser({
+				id: currentUser.uid,
+				...userSnapshot.data(),
+			});
+
+			const pathReference = ref(
+				storage,
+				`usersProfilePhoto/${currentUser.uid}.jpg`
+			);
+			getDownloadURL(pathReference)
+				.then((url) => {
+					setUserProfileImgURL(url);
+				})
+				.catch((error) => {
+					console.log(error);
```

Comment:
Może zamienić to na jakiegoś toast - nawet po prostu z informacją "Wystąpił błąd serwera" czy coś w tym rodzaju?

Reply:
poprawione na toast